client/layout/controller.js

```diff
@@ -213,52 +205,22 @@ export class Controller extends Component {
 	}
 }
 
-/**
- * Update an anchor's link in sidebar to include persisted queries. Leave excluded screens
- * as is.
- *
- * @param {HTMLElement} item - Sidebar anchor link.
- * @param {Object} nextQuery - A query object to be added to updated hrefs.
- * @param {Array} excludedScreens - wc-admin screens to avoid updating.
- */
-export function updateLinkHref( item, nextQuery, excludedScreens ) {
-	if ( isWCAdmin( item.href ) ) {
-		const search = last( item.href.split( '?' ) );
-		const query = parse( search );
-		const path = query.path || 'homescreen';
-		const screen = path.replace( '/analytics', '' ).replace( '/', '' );
-
-		const isExcludedScreen = excludedScreens.includes( screen );
-
-		const href =
-			'admin.php?' +
-			stringify(
-				Object.assign( query, isExcludedScreen ? {} : nextQuery )
-			);
-
-		// Replace the href so you can see the url on hover.
-		item.href = href;
+// Add click handlers to wc-admin links in wp-admin menu to suppress page refresh when possible
+const patchMenuLinks = function () {
+	Array.from( document.querySelectorAll( '#adminmenu a' ) ).forEach(
+		( item ) => {
+			if ( ! isWCAdmin( item.href ) ) {
+				return;
+			}
 
-		item.onclick = ( e ) => {
-			e.preventDefault();
-			getHistory().push( href );
-		};
-	}
-}
-
-// Update's wc-admin links in wp-admin menu
-window.wpNavMenuUrlUpdate = function ( query ) {
-	const excludedScreens = applyFilters( TIME_EXCLUDED_SCREENS_FILTER, [
-		'stock',
-		'settings',
-		'customers',
-		'homescreen',
-	] );
-	const nextQuery = getPersistedQuery( query );
-
-	Array.from(
-		document.querySelectorAll( '#adminmenu a' )
-	).forEach( ( item ) => updateLinkHref( item, nextQuery, excludedScreens ) );
+			item.onclick = ( e ) => {
+				e.preventDefault();
```

Comment:
I wonder if this interferes with the `onClick` on `NavigationItem` we use for Tracks?

Reply:
This part actually isn't new, just relocated from the now deleted `updateLinkHref()` function: https://github.com/woocommerce/woocommerce-admin/blob/67515d8aa6cca68c5ea3171077d4aa38502e52b1/client/layout/controller.js#L242

So I believe it shouldn't behave any differently for tracks/etc?